index.js

```diff
@@ -54,11 +54,11 @@ module.exports = function (obj, opts) {
 			throw new TypeError('Converting circular structure to JSON');
 		} else { seen.push(node); }
 
-		var keys = objectKeys(node).sort(cmp && cmp(node));
+		var keys = objectKeys(node).sort(cmp && (a, b) => cmp(node)(a, b, path));
```

Comment:
arrow function syntax is unavailable
```suggestion
		var keys = objectKeys(node).sort(cmp && function (a, b) { return cmp(node)(a, b, path); });
```